MBProgressHUD.m

```diff
@@ -699,22 +699,28 @@ - (void)updateForCurrentOrientationAnimated:(BOOL)animated {
     BOOL iOS8OrLater = kCFCoreFoundationVersionNumber >= kCFCoreFoundationVersionNumber_iOS_8_0;
     if (iOS8OrLater || ![self.superview isKindOfClass:[UIWindow class]]) return;
 
-    UIInterfaceOrientation orientation = [UIApplication sharedApplication].statusBarOrientation;
-    CGFloat radians = 0;
-    if (UIInterfaceOrientationIsLandscape(orientation)) {
-        radians = orientation == UIInterfaceOrientationLandscapeLeft ? -(CGFloat)M_PI_2 : (CGFloat)M_PI_2;
-        // Window coordinates differ!
-        self.bounds = CGRectMake(0, 0, self.bounds.size.height, self.bounds.size.width);
-    } else {
-        radians = orientation == UIInterfaceOrientationPortraitUpsideDown ? (CGFloat)M_PI : 0.f;
-    }
-
-    if (animated) {
-        [UIView animateWithDuration:0.3 animations:^{
+    // Make extension friendly
+    Class UIApplicationClass = NSClassFromString(@"UIApplication");
+    if(UIApplicationClass && [UIApplicationClass respondsToSelector:@selector(sharedApplication)]) {
```

Comment:
Negating this check and instead exiting early (like in the `iOS8OrLater` check) would be a bit nicer as we would avoid nesting all that code. 

A short comment explaining why we are doing this would also be useful.